parcels/kernels/advectiondiffusion.py

```diff
@@ -0,0 +1,230 @@
+"""Collection of pre-built advection-diffusion kernels"""
+import math
+
+from parcels import rng as random
+
+
+__all__ = ['DiffusionUniformKh', 'AdvectionDiffusionEuler', 'AdvectionDiffusionMilstein1',
+           'AdvectionRK4DiffusionEuler', 'AdvectionRK4DiffusionMilstein1']
+
+
+def DiffusionUniformKh(particle, fieldset, time):
+    """Kernel for diffusion where diffusivity (Kh) is assumed uniform.
+    Assumes that fieldset has fields Kh_zonal and Kh_meridional
+
+    This kernel neglects gradients in the diffusivity field and is
+    therefore more efficient in cases with constant diffusivity.
+    Since the perturbation due to diffusion is in this case spatially
+    independent, this kernel contains no advection and can be used in
+    combination with a seperate advection kernel.
+
+    The Wiener increment `dW` should be normally distributed with zero
+    mean and a standard deviation of sqrt(dt). Instead, here a uniform
+    distribution with the same mean and std is used for efficiency and
+    to keep random increments bounded. This substitution is valid for
+    the convergence of particle distributions. If convergence of
+    individual particle paths is required, use normally distributed
+    random increments instead. See Gräwe et al (2012)
+    doi.org/10.1007/s10236-012-0523-y for more information.
+    """
+    # Wiener increment with zero mean and std of sqrt(dt)
+    dWx = random.uniform(-1., 1.) * math.sqrt(math.fabs(particle.dt) * 3)
+    dWy = random.uniform(-1., 1.) * math.sqrt(math.fabs(particle.dt) * 3)
+
+    bx = math.sqrt(2 * fieldset.Kh_zonal[time, particle.depth, particle.lat, particle.lon])
+    by = math.sqrt(2 * fieldset.Kh_meridional[time, particle.depth, particle.lat, particle.lon])
+
+    particle.lon += bx * dWx
+    particle.lat += by * dWy
+
+
+def AdvectionDiffusionEuler(particle, fieldset, time):
+    """Kernel for simple advection-diffusion, solved using the Euler-Maruyama
```

Comment:
But confusing that Kernel name is `AdvectionDiffusionEuler`; wouldn't `AdvectionDiffusionEulerMaruyama` be clearer?

Reply:
In the literature it's referred to both as Euler and as Euler-Maruyama. I think the Maruyama part stresses that we're dealing with SDEs, but I figured Euler in the kernel name is nice and short...

Reply:
OK, let' leave as is then

Reply:
I'll rename it to EM